### PR TITLE
Fix issue #82

### PR DESCRIPTION
**Pull request by ChromaCat248**

### CODE REVIEW

File: Fushigi/ui/SceneObjects/bgunit/BGUnitRailSceneObj.cs [hunk 1]
```diff
@@ -186,7 +188,7 @@ public void OnMouseDown(CourseAreaEditContext ctx, LevelViewport viewport)
                 }
 
                 DeselectAll(ctx);
-                InsertPoint(ctx, new BGUnitRail.RailPoint(rail, pos), min.index);
+                InsertPoint(ctx, new BGUnitRail.RailPoint(rail, pos), rail.Points.Count);
```
Comment (jupahe64) [inline Fushigi/ui/SceneObjects/bgunit/BGUnitRailSceneObj.cs:191]: why is this change necessary to fix the issue, the whole point of the code above is to calculate min.index.
if you think that is not how it should be done we can discuss this but don't just make a change that's unrelated to the issue.

Comment (ChromaCat248) [reply Fushigi/ui/SceneObjects/bgunit/BGUnitRailSceneObj.cs]: i made that change because `min.index` was always 1 which was causing an error since an index of 1 is outside the bounds for an empty list, upon further inspection it only did that because my fix was causing the if statement on line 172 to set `min.index` to 1

i've pushed a fix that fixes my fix
